astroquery/simbad/deprecated_core.py

```diff
@@ -0,0 +1,1160 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+The deprecated interface to query Simbad.
+
+The features in this interface should be available in the new one. Indications on how to switch to the new
+interface are listed at `https://astroquery.readthedocs.io/en/stable/simbad/simbad_evolution.html`_.
+However if something is missing or unclear, don't hesitate to send an email to cds-question[at]unistra.fr or
+open an issue in astroquery's github.
+"""
+
+import re
+import requests
+import json
+import os
+from collections import namedtuple
+from io import BytesIO
+import warnings
+import astropy.units as u
+from astropy.utils import isiterable
+from astropy.utils.data import get_pkg_data_filename
+import astropy.coordinates as coord
+from astropy.table import Table
+import astropy.io.votable as votable
+
+from astroquery.query import BaseQuery
+from astroquery.utils import commons, async_to_sync
+from astroquery.exceptions import TableParseError, LargeQueryWarning, BlankResponseWarning
+
+from . import conf
+
+
+__all__ = ['DeprecatedSimbad', 'DeprecatedSimbadClass', 'DeprecatedSimbadBaseQuery']
```

Comment:
I'm not sure that we need to keep these classes. 

The usual purpose of deprecation is to give some time for users to update their code when the API changes. With this PR, the astroquery.simbad.Simbad class already changes and even those who would want to fall back to the previous behaviour will need to update their code to use the different, deprecated classes.

Reply:
Hi, my position is a bit more in favor of having this "deprecatedSimbad", but if not, that's ok too. 
Just to explain our goal with this deprecated class: the idea is to allow people to upgrade the astroquery module (for reasons other than Simbad) and if they find issues with this new behavior, it would be much faster to simply change the module name rather than deal with all the changes (not so much, but...). Apart from that, I completely agree that it is not usual, nor good practice, it was just a way of being more smooth for the user.

Reply:
Is there a project for an astroquery v5? Maybe this could fit into a major number bump? And then we'd feel less bad about changing a bit the API?